File: task3.exs
defmodule Task3 do
  def proverka(n) do
    digits = Integer.digits(n)
    sum = Enum.sum(digits)
    proizv = Enum.product(digits)
    proizv > sum
  end

  def main(args) do
    IO.puts("введите N:")
    n = IO.read(:line) |> String.trim() |> String.to_integer()
    IO.puts("введите N  чисел:")
    numbers = for _ <- 1..n do
      IO.read(:line) |> String.trim() |> String.to_integer()
    end
    result = Enum.filter(numbers, &proverka/1)
    output = if Enum.empty?(result), do: "не найдено", else: Enum.join(result, " ")
    IO.puts(output)

    """
    {input, output} = if "--file" in args do
      {:ok, f_in} = File.open("input.txt", [:read])
      {:ok, f_out} = File.open("output.txt", [:write])
      {f_in, f_out}
    else
      {:stdio, :stdio}
    end
    n = IO.read(input, :line) |> String.trim() |> String.to_integer()
    numbers = for _ <- 1..n do
      IO.read(input, :line) |> String.trim() |> String.to_integer()
    end
    result = Enum.filter(numbers, &proverka/1)
    output = if Enum.empty?(result), do: "не найдено", else: Enum.join(result, " ")
    IO.puts(output, output)
    if input != :stdio, do: File.close(input)
    if output != :stdio, do: File.close(output)
    """
  end
end

System.argv() |> Task3.main()

File: task3.cpp
#include <iostream> 
#include <fstream>
#include <vector>

using namespace std;

bool proverka(int x) {
    int sum = 0, proizv = 1;
    int n = x;
    while (n > 0) {
        int d = n % 10;
        sum += d;
        proizv *= d;
        n /= 10;
    }
    return proizv > sum;
}

int main() {
    int N;
    cout << "введите N: ";
    cin >> N;
    vector<int> ans;
    cout << "введите " << N << " чисел: ";
    for (int i = 0; i < N; ++i) {
        int x;
        cin >> x;
        if (proverka(x)) {
            ans.push_back(x);
        }
    }

    if (ans.empty()) {
        cout << "не найдено" << endl;
    } else {
        for (size_t i = 0; i < ans.size(); ++i) {
            cout << ans[i] << " ";
        }
        cout << endl;
    }

    /*
    ifstream fin("input.txt");
    ofstream fout("output.txt");
    int N;
    fin >> N;
    vector<int> ans;
    for (int i = 0; i < N; ++i) {
        int x;
        fin >> x;
        if (proverka(x)) ans.push_back(x);
    }
    if (ans.empty()) fout << "не найдено" << endl;
    else {
        for (int x : ans) fout << x << " ";
        fout << endl;
    }
    fin.close();
    fout.close();
    */

    return 0;
}
